### PR TITLE
fix: reenable the note block and picker field

### PR DESCRIPTION
**Pull request by gonfunko**
This PR reenables the note block and the note picker (piano) field.

### CODE REVIEW

File: core/field_note.js [hunk 1]
```diff
@@ -42,809 +35,797 @@ goog.require('goog.userAgent');
...
-  this.octaveUpMouseDownWrapper_ =
-      Blockly.bindEvent_(this.octaveUpButton, 'mousedown', this,function() {
-        this.changeOctaveBy_(1);
+  /**
+   * Show a field with piano keys.
+   * @private
+   */
+  showEditor_(event) {
+    // Mobile browsers have issues with in-line textareas (focus & keyboards).
+    super.showEditor_(event, this.useTouchInteraction_);
```
Comment (BeksOmega) [inline core/field_note.js]: I don't think `useTouchInteraction_` works anymore.

I think you should take in a `quietInput` parameter and pass that to `super`: https://github.com/google/blockly/blob/5a5184ab4fa911b7826d5b424716b972215b1740/core/field_input.ts#L324

File: core/field_note.js [hunk 2]
```diff
@@ -42,809 +35,797 @@ goog.require('goog.userAgent');
...
+
+    // If there is an existing drop-down someone else owns, hide it immediately and clear it.
+    Blockly.DropDownDiv.hideWithoutAnimation();
+    Blockly.DropDownDiv.clearContent();
```
Comment (BeksOmega) [inline core/field_note.js]: I don't think you actually need to do this. Could you try deleting it?

File: core/field_note.js [hunk 3]
```diff
@@ -42,809 +35,797 @@ goog.require('goog.userAgent');
...
+    this.htmlInput_.value = noteNum;
+
+    if (!this.textElement_) {
+      // Not rendered yet.
+      return;
+    }
+    this.updateSelection_();
+    // Cached width is obsolete.  Clear it.
+    this.size_.width = 0;
+  }
```
Comment (BeksOmega) [inline core/field_note.js]: Can you remove this and just have it use `setValue` + override the other value methods? I think that should be workable.

**Review by BeksOmega — CHANGES_REQUESTED** [review]
The SVG code in this is kind of crazy haha. Just a few fixes.

File: core/field_note.js (hunk 1 above)
Comment (gonfunko) [reply core/field_note.js]: Done

File: core/field_note.js (hunk 2 above)
Comment (gonfunko) [reply core/field_note.js]: Done

File: core/field_note.js (hunk 3 above)
Comment (gonfunko) [reply core/field_note.js]: Done, had to use `setEditorValue_` to have the change reflected in the block though.

Comment (BeksOmega) [reply core/field_note.js]: Yeah that should be correct! I disagreed with that design decision, but it is how it works now :P lmk if it needs doc updates.

**Review by BeksOmega — APPROVED** [review]

**Pull request merged**